### PR TITLE
feat:check if editor is spammer before inserting URL in internet_archive_urls

### PR DESCRIPTION
**Pull request by yellowHatpro**
- We check if any editor is spammer or not each time we try to extract URL from `edit_note` and `edit_data`

### CODE REVIEW

File: src/poller/utils.rs [hunk 1]
```diff
@@ -232,7 +301,7 @@ pub async fn save_url_to_internet_archive_urls(
     from_table: &str,
     from_table_id: i32,
     pool: &PgPool) {
-    if should_insert_url_to_internet_archive_urls(url, pool).await.expect("Error: ") {
+    if let Ok(_)= should_insert_url_to_internet_archive_urls(url, pool).await {
```
Comment (RustyNova016) [inline src/poller/utils.rs:304]: Sorry for the backseat review, but this line seems really suspicious in both logic and style. Here's a few things that are wrong IMO

- The condition is equivalent to checking if the return value of the function is `Ok` or `Err`. It doesn't check the returned boolean. You need to unwrap the result first to get the boolean. If you are actually checking whether the result "is ok", prefer using `Result::is_ok()` as it's more readable
- You are discarding the `Err` value silently. Usually you should explicitly deal with the `Err` value by either doing `.expect()`, `match` or just sending it to the return value with `?`. 
- If you don't care about the `Err` value, it's recommended to put a comment to tell why this is fine
- The formating isn't correct. IDK what policy there is for formating, but you can use `cargo fmt` to easily format your code

Comment (yellowHatpro) [reply src/poller/utils.rs]: Hi @RustyNova016 thanks a lot for the review, I forgot that we were returning `bool` wrapped Result from the function `should_insert_url_to_internet_archive_urls` and assumed it was fine to check if the result is `Ok`. 
Also regarding the Err, I think we should generalize this for all util functions that returns Result.
- Cooool, I think I should automate cargo fmt some how on each commit, that would help format it before pushing
Thanks a ton, it really helped a lot 😁

File: src/poller/utils.rs [hunk 2]
```diff
@@ -231,23 +288,29 @@ pub async fn save_url_to_internet_archive_urls(
...
+                    .execute(pool)
+                    .await
+                    .unwrap();
+            }
+        }
+        Err(e) => {
+            eprintln!("Error saving {} into internet_archive_urls: {}", url, e)
+        }
     }
 }
```
Comment (yellowHatpro) [inline src/poller/utils.rs:312]: @RustyNova016 updated the logic here

File: src/poller/utils.rs (hunk 1 above)
Comment (RustyNova016) [reply src/poller/utils.rs]: Glad to help and finally contribute to an Metabrainz project. Might make some PR to fix some parts to make them more idiomatic. I suppose you're a quite new rustacean so it's understandable 

Although I did notice that your fix commit included the whole reformating of the app. Maybe do it a separate PR? Other reviewer may find it hard to review the PR now that it is filed with formatting changes.

As for auto reformating, I don't know how to do it outside of Rust Rover. But I do know how to use CI to refuse unformatted PRs.

Comment (RustyNova016) [reply src/poller/utils.rs]: Exemple here
https://github.com/RustyNova016/listenbrainz-cli-tools/blob/master/.github%2Fworkflows%2Frust.yml#L116

File: src/poller/utils.rs (hunk 2 above)
Comment (RustyNova016) [reply src/poller/utils.rs:312]: Looks good to me. 👍

File: src/poller/utils.rs (hunk 1 above)
Comment (yellowHatpro) [reply src/poller/utils.rs]: Thanks, will update the PR soon, will get rid of the latest commit and update with newer one with  changes only in `save_url_to_internet_archive_urls` function.
Regarding the CI, I haven't added any CI pipeline yet, but will have a look at the example and add a workflow as well. ✅

File: src/poller/utils.rs (hunk 2 above)
Comment (yellowHatpro) [reply src/poller/utils.rs:312]: @RustyNova016 btw, should I try to `Return<(), Error>` in this case?
I think in general, in functions related to db access, I should do the same. Thoughts?

Comment (RustyNova016) [reply src/poller/utils.rs]: Here's some rules of thumb I try to follows:
- Errors that I don't want to deal here (ex: not enough context about the caller)? Return the error
- Got an error that you know you can handle here and the caller isn't a better place to handle? Handle the error state here
- Got an error that you know is impossible to happen (like inserting in a Vec then getting the first element)? Use `.except()` and provide an explanation of why this isn't an issue, and if that ever happens, some clues as to why it may happen.

In your case, just return `Result<>`. But this isn't a change in a vacuum. IMO removing all those unwraps and setting function returns should be it's own PR, but I'm not your project manager so ehh

Comment (yellowHatpro) [reply src/poller/utils.rs]: Thanks, I will come up with a cleanup and refactoring PRs for this.
I didn't really focus much on doing things idiomatic way, rather exploring different ways to do them.
Thanks to your PR, I am getting a lot of sense of clean code, and will try to incorporate the coding style in future contributions ✅.

File: src/poller/utils.rs [hunk 3]
```diff
@@ -106,6 +126,50 @@ fn any_annotation(json: &JsonValue) -> Option<Vec<String>> {
...
+    "#;
+
+    let res = sqlx::query_as::<_, (i16, )>(query)
+        .bind(edit_id)
+        .fetch_one(pool)
+        .await;
+    match res {
+        Ok((id, )) => { Some(id) }
+        Err(e) => {
+            eprintln!("error: {}", e);
```
Comment (RustyNova016) [inline src/poller/utils.rs:167]: Instead of printing and returning none, it might be better to actually handle the error case outside of the function. 
But might be better for another PR

**Review by RustyNova016 — COMMENTED** [review]
Finally got enough time for a full review.
I didn't test it so I can't vouch for it actually working, but this seems fine

File: src/poller/utils.rs (hunk 3 above)
Comment (yellowHatpro) [reply src/poller/utils.rs:167]: Cool I'll return a Result wrapped later then

File: src/poller/utils.rs [hunk 4]
```diff
@@ -1,10 +1,37 @@
...
+
+/// This function takes input Edit Data, checks if
+/// - the Edit Data contains URL
+/// - the Editor is not a spammer
+///
+/// and returns the vector of (URL as String)
+pub async fn extract_url_from_edit_data(edit: &EditData, pool: &PgPool) -> Vec<String> {
+    let json: &Value = &edit.data;
+    if let Some(edit_type_if_editor_not_spammer) = get_edit_type_if_editor_is_not_spammer(edit.edit, pool).await {
+        extract_urls_from_json(json, edit_type_if_editor_not_spammer.into())
```
[the remark this answers is not in the document]
Comment (yellowHatpro) [reply src/poller/utils.rs:29]: Oof, just enabled Clippy in RustRover, I can see what I couldn't earlier.

**Comment by yellowHatpro** [comment]
I have made the necessary changes.
I will make a dedicated PR for all the remaining linting and refactoring work, since I want to keep this PR limited to spam edit logic. 
Thanks @RustyNova016 for the help 🙌

**Comment by yellowHatpro** [comment]
Rebasing it rn in favour of new PR for refactoring of codebase with better rust standards

File: src/poller/utils.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (RustyNova016) [reply src/poller/utils.rs]: Don't forget to ingrave the reformat file shortcut in your muscle memory. It's also super useful. Should be ctrl+shift+alt+L.
